src/strands/hooks/interrupt.py

```diff
@@ -0,0 +1,104 @@
+"""Human-in-the-loop interrupt system for agent workflows."""
+
+from dataclasses import asdict, dataclass
+from typing import TYPE_CHECKING, Any
+
+from ..types.tools import ToolResultContent
+
+if TYPE_CHECKING:
+    from ..agent import Agent
+
+
+@dataclass
+class Interrupt:
```

Comment:
Is this interrupt going to be used outside of hooks?  If so, I don't think it belongs under the hooks module